src/frequenz/sdk/microgrid/_battery/_status.py

```diff
@@ -0,0 +1,373 @@
+        # every call.
+        # pylint: disable=attribute-defined-outside-init
+        self._battery_receiver = _ComponentReceiver(
+            self._battery_id, bat_recv.into_peekable()
+        )
+        inv_recv = await api_client.inverter_data(inverter_id)
+        # pylint: disable=attribute-defined-outside-init
+        self._inverter_receiver = _ComponentReceiver(
+            inverter_id, inv_recv.into_peekable()
+        )
+
+        await asyncio.sleep(_START_DELAY_SEC)
+        self._init_method_called = True
+
+    def get_status(self) -> BatteryStatus:
+        """Return status of the battery.
+
+        The decision is made based on last message from battery and adjacent inverter
+        and result from last request.
+
+        Raises:
+            RuntimeError: If method `async_init` was not called or not awaited.
+
+        Returns:
+            Battery status
+        """
+        if not self._init_method_called:
+            raise RuntimeError(
+                "`async_init` method not called or not awaited. Run it before first use"
+            )
+
+        bat_msg = self._battery_receiver.receiver.peek()
+        inv_msg = self._inverter_receiver.receiver.peek()
+        if bat_msg is None or inv_msg is None:
+            if self._last_status == BatteryStatus.WORKING:
+                if not bat_msg:
+                    component_id = self._battery_id
+                else:
+                    component_id = self._inverter_receiver.component_id
+
+                _logger.warning(
+                    "None returned from component %d receiver.", component_id
+                )
+            self._last_status = BatteryStatus.NOT_WORKING
+            return self._last_status
+
+        older_message = bat_msg if bat_msg.timestamp < inv_msg.timestamp else inv_msg
+        is_msg_ok = (
+            self._is_message_reliable(older_message)
+            and self._is_battery_state_correct(bat_msg)
+            and self._is_inverter_state_correct(inv_msg)
+            and self._no_critical_error(bat_msg)
+            and self._no_critical_error(inv_msg)
+        )
+
+        if not is_msg_ok:
+            self._last_status = BatteryStatus.NOT_WORKING
+            return self._last_status
+
+        # Use battery as soon as its message is back correct.
+        if self._last_status == BatteryStatus.NOT_WORKING:
+            self.unblock()
+
+        if self.is_blocked():
+            self._last_status = BatteryStatus.UNCERTAIN
+        else:
+            self._last_status = BatteryStatus.WORKING
+
+        return self._last_status
+
+    def is_blocked(self) -> bool:
+        """Return if battery is blocked.
+
+        Battery can be blocked if last request for that battery failed.
+
+        Returns:
+            True if battery is blocked, False otherwise.
+        """
+        if self._blocked_until is None:
+            return False
+        return self._blocked_until > datetime.now(tz=pytz.UTC)
```

Comment:
Nitpick: I would avoid using `pytz` only for `UTC`, as it is available in the stdlib as `datetime.timezone.utc`.

Reply:
Done